dvc/repo/remove.py

```diff
@@ -11,9 +11,16 @@
 
 @locked
 def remove(self: "Repo", target: str, outs: bool = False):
-    stages = self.stage.from_target(target)
+    stages_info = self.stage.collect_granular(target)
 
-    for stage in stages:
-        stage.remove(remove_outs=outs, force=outs)
+    for stage_info in stages_info:
```

Comment:
Let's just unpack them here, makes it easier.
```suggestion
    for stage, filter_info in stages_info:
```